index.js

```diff
@@ -28,6 +31,13 @@ module.exports = (nextConfig = {}) => {
               fallback: require.resolve("file-loader"),
               publicPath: `${nextConfig.assetPrefix || nextConfig.basePath}/_next/static/images/`,
               outputPath: `${isServer ? "../" : ""}static/images/`,
+              postTransformPublicPath: (p) => {
+                if (!nextConfig.assetPrefix) {
+                  return `(require("next/config").default().publicRuntimeConfig.nextImagesAssetPrefix || '') + ${p}`
```

Comment:
@peter-jozsa Opened this PR in my editor, this line doesn't work, because you need to put the require expression in curly braces to make it get interpreted:
```javascript
return `${require("next/config").default().publicRuntimeConfig.nextImagesAssetPrefix || ''} + ${p}`
```
Or you can also make it prettier this way:
```javascript
(require("next/config").default().publicRuntimeConfig.nextImagesAssetPrefix || '').concat(p)
```
I also think you can read publicRuntimeConfig from the nextConfig object, just like what you did in line 3. So we can have this:
```javascript
(nextConfig. publicRuntimeConfig || "").concat(p)
```

Reply:
If you put it in curly braces like you did it will be interpreted during build time, if you use it as a string the given expression will be evaluated during runtime. Build time evaluation won't help us regarding runtime asset prefixes.

Reply:
@peter-jozsa Ah I see, you're right. If you want, you can also add the readme section, else I'll merge this PR and then will add the readme and bump the version up.

Reply:
I was just about to respond to your comment. I'm afraid of releasing this feature in this form. I think we should provide a configuration variable to let users decide if they want to use this feature. Currently it will import `next/config` for every user who update to this version.

Reply:
@peter-jozsa Ah sorry! Yeah, I agree... Can you please open another PR for that and also the readme file?

Reply:
Currently this package doesn't have its own configuration variables. What do you think how should we introduce it? I think it should be a backward-compatible change so no major version change is required. 

We could add a new parameter to `withImages` like:
```js
withImages(nextConfig, { dynamicAssetPrefix: true })
```

or we could add it to existing nextConfig parameter, like:
```js
withImages({
  assetPrefix: '....',
  webpack () { ... },
  dynamicAssetPrefix: true
})
```

Personally I like the second one since we could clean `dynamicAssetPrefix` from `nextConfig` variable before returning it.

Reply:
Yeah I'll also go with the second option 👍